lib/valueflow.cpp

```diff
@@ -263,7 +263,11 @@ static ValueFlow::Value castValue(ValueFlow::Value value, const ValueType::Sign
 {
     if (value.isFloatValue()) {
         value.valueType = ValueFlow::Value::INT;
-        value.intvalue = value.floatValue;
+        if (std::numeric_limits<int>::min() <= value.floatValue && value.floatValue <= std::numeric_limits<int>::max()) {
```

Comment:
this condition is weird for me as a C++ developer. I assume you are influenced by python 1<x<5 syntax. please put value.floatValue in both lhs.